### DISCUSSION TITLE
Check if a map contains a key?

### DISCUSSION THREAD

**Discussion by daandemeyer** [bpftrace/bpftrace#2054] [Q&A] [answered]
Is there a way to check if a map contains a specific key? I looked at the reference guide but couldn't find an example

**Comment by fbs** [comment, marked as answer]
```
kprobe:x 
/@map[key]/
{
  do something conditionally
```

something like this should work

**Reply by ulissesf** [reply]
I just checked that if the value stored in @map[key] is a tuple it doesn't work, though. Any way to do that then? Thanks.

**Reply by yump** [reply]
I ran into this problem and solved it by checking `@Map[key].0 != 0`.  Of course that's only possible if your tuple has at least one element where 0 isn't a valid value.
